src/react-hooks/buildHooks.ts

```diff
@@ -262,6 +344,24 @@ export function buildHooks<Definitions extends EndpointDefinitions>({
     return {
       useQueryState,
       useQuerySubscription,
+      useLazyQuerySubscription,
+      useLazyQuery(options) {
+        const [providedArgs, setProvidedArgs] = useState(null);
+        const [trigger] = useLazyQuerySubscription(options);
+        const queryStateResults = useQueryState(providedArgs, {
+          ...options,
+          skip: providedArgs === null,
+        });
+        const triggerQuery = useCallback(
+          function (args: any) {
+            setProvidedArgs(args);
+            trigger(args);
+          },
+          [trigger]
+        );
+
+        return useMemo(() => [triggerQuery, queryStateResults], [triggerQuery, queryStateResults]);
```

Comment:
I kinda also want to expose the last args, so maybe

```suggestion
        const info = useMemo(() => ({ lastArgs: promise.args }, [ promise.args ]);
        return useMemo(() => [triggerQuery, queryStateResults, info], [triggerQuery, queryStateResults, info]);
```
?